File: src/main/java/com/tealcube/minecraft/bukkit/mythicdrops/socketting/SocketItem.java
package com.tealcube.minecraft.bukkit.mythicdrops.socketting;

/*
 * #%L
 * MythicDrops
 * %%
 * Copyright (C) 2013 - 2015 TealCube
 * %%
 * Permission to use, copy, modify, and/or distribute this software for any purpose with or without fee is hereby granted,
 * provided that the above copyright notice and this permission notice appear in all copies.
 * 
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
 * INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF
 * THIS SOFTWARE.
 * #L%
 */


import com.tealcube.minecraft.bukkit.mythicdrops.MythicDropsPlugin;
import com.tealcube.minecraft.bukkit.mythicdrops.api.items.MythicItemStack;
import com.tealcube.minecraft.bukkit.mythicdrops.utils.StringListUtil;
import com.tealcube.minecraft.bukkit.mythicdrops.utils.StringUtil;

import org.bukkit.Material;
import org.bukkit.material.MaterialData;

public final class SocketItem extends MythicItemStack {

    public SocketItem(Material material, SocketGem socketGem) {
        super(material, 1, (short) 0, StringUtil.replaceArgs(MythicDropsPlugin
                                                                 .getInstance()
                                                                 .getSockettingSettings()
                                                                 .getSocketGemName(),
                                                             new String[][]{
                                                                 {"%socketgem%",
                                                                  socketGem
                                                                      .getName()}}
              ),
              StringListUtil.replaceArgs(MythicDropsPlugin.getInstance()
                                             .getSockettingSettings().getSocketGemLore(),
                                         new String[][]{{"%type%", socketGem.getPresentableType()}}
              )
        );
    }

    @Deprecated
    public SocketItem(MaterialData materialData, SocketGem socketGem) {
        super(materialData.getItemType(), 1, (short) 0, StringUtil.replaceArgs(MythicDropsPlugin
                                                                                   .getInstance()
                                                                                   .getSockettingSettings()
                                                                                   .getSocketGemName(),
                                                                               new String[][]{
                                                                                   {"%socketgem%",
                                                                                    socketGem
                                                                                        .getName()}}
              ),
              StringListUtil.replaceArgs(MythicDropsPlugin.getInstance()
                                             .getSockettingSettings().getSocketGemLore(),
                                         new String[][]{{"%type%", socketGem.getPresentableType()}}
              )
        );
    }

}

File: src/main/kotlin/com/tealcube/minecraft/bukkit/mythicdrops/socketting/SocketItem.kt
/*
 * The MIT License
 * Copyright © 2013 Richard Harrah
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package com.tealcube.minecraft.bukkit.mythicdrops.socketting

import com.tealcube.minecraft.bukkit.mythicdrops.MythicDropsPlugin
import com.tealcube.minecraft.bukkit.mythicdrops.replaceArgs
import io.pixeloutlaw.minecraft.spigot.hilt.setDisplayName
import io.pixeloutlaw.minecraft.spigot.hilt.setLore
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class SocketItem(material: Material, socketGem: SocketGem) : ItemStack(material, 1) {
    init {
        this.setDisplayName(
                MythicDropsPlugin.getInstance()
                        .sockettingSettings
                        .socketGemName
                        .replaceArgs(
                                "%socketgem%" to socketGem.name
                        )
        )
        this.setLore(
                MythicDropsPlugin.getInstance()
                        .sockettingSettings
                        .socketGemLore
                        .replaceArgs(
                                "%type%" to socketGem.presentableType
                        )
        )
    }
}